src/main/java/frc/robot/RobotContainer.java

```diff
@@ -152,6 +157,14 @@ private void setSpeedFactor(double speedFactor) {
     m_speedFactor = speedFactor;
   }
 
+  private boolean withinTolerance(Pose2d referencePose, Pose2d pose, double xTolerance, double yTolerance) {
+    Transform2d posesDifference = referencePose.minus(pose);
```

Comment:
This would all be simpler (and IMO better) if based on a single tolerance rather than x,y components.

```java
  return referencePose.minus(pose).getNorm() < tolerance;
```